src/gui/mainwindow.cpp

```diff
@@ -148,6 +149,8 @@ MainWindow::MainWindow(IGUIApplication *app, const WindowState initialState, con
     // Setting icons
 #ifndef Q_OS_MACOS
     setWindowIcon(UIThemeManager::instance()->getIcon(u"qbittorrent"_s));
+#else // Q_OS_MACOS
+    MacUtils::askForNotificationPermission();
```

Comment:
The reason I suggested to move to the bottom is because here (the top of the constructor) is the code that initialize/setup this class for it to run properly. The `MacUtils::askForNotificationPermission` doesn't seems like a strict requirement for `MainWindow`. It seems to be more like hitchhiking on this class. In this case, it would be better to move it to the end/bottom of the constructor since it is only loosely related to this class.

Reply:
Isn't `DesktopIntegration` better place for it?